commons/src/main/java/org/infinispan/commons/marshall/StreamAwareMarshaller.java

```diff
@@ -0,0 +1,61 @@
+package org.infinispan.commons.marshall;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import org.infinispan.commons.dataconversion.MediaType;
+
+import net.jcip.annotations.ThreadSafe;
+
+/**
+ * An interface that facilitates the marshalling/unmarshalling of objects from
+ * the provided {@link java.io.OutputStream}/{@link java.io.InputStream}.
```

Comment:
> And since the package is `@public`, do we expect users to use or implement it?

I think this depends on whether we want users to be able to provide their own marshallers or not as we discussed on Zulip. If we're going to continue to support this, then I think we should make this interface public and make it the preferred interface for users to implement as it's the most simple. However, if we're going to deprecate custom user marshallers then I don't see any reason for it to be public as the `InitializationContext` exposes the `PersistenceMarshaller` interface which extends `StreamAwareMarshaller`.

Internally, we should be aiming to use `StreamAwareMarshaller` and long term we should try to avoid calls to `objectToByteBuffer` etc where possible.

Reply:
@ryanemerson because of the `@public` tag in package-info.java, all the classes in `org.infinispan.commons.marshall` are already public, so you should add an explicit `@private` tag if you don't want it to be public (yet). Alternatively you could remove the `@public` tag from the package and add it only to the classes which really need to be public.